### DISCUSSION TITLE
UPBGE 0.3 - Feature request: Action Actuator Layers blend-out

### DISCUSSION THREAD

**Discussion by arch-sketch** [UPBGE/upbge#1462] [Ideas/Features for UPBGE 0.3+]
Hi, I was searching online for more info on this and found this forum post that discusses this issue exactly:

https://blenderartists.org/t/action-layers-blending-out/547964

Quoting: "When I play more than one action simultaneously on different layers, using blendin works fine at the beginning of an action, but is no good when the action on top ends and the action underneath resumes - it jerks back to the other action.
For example, when a character is running and he starts an attack animation on a new layer with a blendin of 5, he transitions smoothly from running to attacking, but as soon as the attacking is over, there is no transition at all back to just running. This seems like a drawback to the current layered action system"

It seems this issue still remains. If you have a walking animation playing (that moves the whole armature) and you start a 'moving arms' animation on top of it in another layer (that moves only the arms bones), there is a smooth transition upon starting, but it simply snaps back to the botton walking animation when the top animation is done. I uploaded an example:

https://we.tl/t-3TpXxgN5iJ

(Press P to play, press spacebar once to go from idle to walking, then spacebar again to raise arms, and you can see the snap back when the arms are lowered at the end of the animation)

In the original post, a solution is discussed: "Since there are two actions playing at the same time on different layers, couldn’t you just gradually diminish the influence of the top action until the action underneath takes over?" 

Appearently he was able to solve it by coding the solution above in a python script, but it seems the logic brick issue was never resolved.
So the feature request would be to have this functionality inside the action actuator, and be able to set the number of frames to transition between an action that just played on a higher layer to an action that is currently playing on a lower layer.

I thought of a workaround that should in theory bypass this problem, but I appear to have found a bug, so I'll make a new post for that.

**Comment by BluePrintRandom** [comment]
I do agree that 'blend out' would be amazing

overall being able to do 

pose3 = pose1.interpolate(pose2, .5) 

would be amazing
